src/utils/DateUtils.js

```diff
@@ -131,22 +128,22 @@ module.exports = {
      * @param {Date}
      * @return {String}
      */
-    formatDate(date, format) {
+    formatDate(date, format, traslation) {
         let year = date.getFullYear()
         let month = date.getMonth() + 1
         let day = date.getDate()
-        let monthName = this.getMonthName(date.getMonth())
+        let monthName = this.getMonthName(date.getMonth(), traslation.months.original)
         let str = format
+            .replace(/S/, this.getNthSuffix(date.getDate()))
+            .replace(/D/, this.getDayNameAbbr(date, traslation.days))
```

Comment:
_Yes, the problem is this._
If you pick **September** or **December** in datepicker and the format is _'d MMMM yyyy'_ your return is **'8 theptember 2016'**, the suffix _/S/_ overwrites 'S'eptember to 'Th'eptember.

![errordatepicker](https://cloud.githubusercontent.com/assets/1901273/16895416/09ab4d80-4b3b-11e6-80a2-dfdb748fe104.JPG)

It's therefore, I changed the elements initially to overwrites those rules first and then the others.